include/revng/AutoEnforcer/AutoEnforcerContext.h

```diff
@@ -0,0 +1,82 @@
+#pragma once
+
+//
+// This file is distributed under the MIT License. See LICENSE.md for details.
+//
+
+#include <initializer_list>
+
+#include "llvm/ADT/ArrayRef.h"
+#include "llvm/ADT/SmallVector.h"
+#include "llvm/ADT/StringMap.h"
+#include "llvm/ADT/StringRef.h"
+#include "llvm/Support/Casting.h"
+
+#include "revng/AutoEnforcer/KindsRegistry.h"
+#include "revng/AutoEnforcer/SavableObject.h"
+
+namespace AutoEnforcer {
+class NamedGlobal {
+public:
+  NamedGlobal(llvm::StringRef Name, SavableObjectBase &Obj) :
+    Name(Name), Obj(&Obj) {}
+
+  llvm::StringRef getName() const { return Name; }
+
+  SavableObjectBase &getObj() const { return *Obj; }
+
+private:
+  llvm::StringRef Name;
+  SavableObjectBase *Obj;
+};
+
+class Context {
+public:
+  explicit Context(KindsRegisty registry) : KRegistry(std::move(registry)) {}
+  Context();
+  Context(llvm::ArrayRef<NamedGlobal> Globals);
```

Comment:
Why do we have two such different constructors such as `Context(KindsRegisty registry)` and `Context(llvm::ArrayRef<NamedGlobal> Globals)`?

What is the exact purpose of `Context`? Holding a list of globals or a  list of kinds?

Reply:
everything that has a lifetime superior to that of the auto-enforcer itself, so kinds and globals. i guess what is really missing is the constructor that accepts both the lists

Reply:
ACK, put some doxygen stating that.
Maybe we can just have a constructor taking both, worst case you pass in empty arguments. Take a look at call sites to evaluate whether this make sense or not.